### DISCUSSION TITLE
Persistent keyboard layout per application

### DISCUSSION THREAD

**Discussion by Vatson112** [YaLTeR/niri#331] [Q&A]
It will be usefull if we can set persistent keyboard layout per application. (by window class or smth)

I tried to write script using `wlrctl` and `setxkbmap`, but  `setxkbmap` dont work on Wayland. Is there a way on Wayland to set keyboard layout from terminal?

Example usecase:

- Change layout to en_EN when switch to terminal
- Change layout to <other_layout> when switch to messanger

**Comment by YaLTeR** [comment]
Have you tried track-layout window? https://github.com/YaLTeR/niri/wiki/Configuration:-Input#layout

**Reply by Vatson112** [reply]
Thank you. Missed it in docs.
